internal/server/helper.go

```diff
@@ -1,33 +1,69 @@
 package server
 
 import (
-	"log"
-
 	"github.com/SOMAS2020/SOMAS2020/internal/common"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/config"
 	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 )
 
-func getClientInfoFromRegisteredClients(registeredClients map[shared.ClientID]common.Client) map[shared.ClientID]common.ClientInfo {
+func getClientInfosAndMapFromRegisteredClients(
+	registeredClients map[shared.ClientID]common.Client,
+) (map[shared.ClientID]common.ClientInfo, map[shared.ClientID]common.Client) {
 	clientInfos := map[shared.ClientID]common.ClientInfo{}
+	clientMap := map[shared.ClientID]common.Client{}
 
 	for id, c := range registeredClients {
 		clientInfos[id] = common.ClientInfo{
-			Client:    c,
-			Resources: common.DefaultResources,
-			Alive:     true,
+			Resources:                    config.InitialResources,
+			Alive:                        true,
+			Critical:                     false,
+			CriticalConsecutiveTurnsLeft: config.MaxCriticalConsecutiveTurns,

```

Comment:
I'm not sure if turns left is really the way to go, i feel like having a critical turns counter makes more sense

Reply:
Cool